dlt/sources/helpers/rest_client/paginators.py

```diff
@@ -869,12 +875,27 @@ def __init__(
 
         self.cursor_param = cursor_param
         self.cursor_body_path = cursor_body_path
+        self.stop_after_empty_page = stop_after_empty_page
+        self.has_more_path = has_more_path
 
     def update_state(self, response: Response, data: Optional[List[Any]] = None) -> None:
         """Extracts the cursor value from the JSON response."""
         values = jsonpath.find_values(self.cursor_path, response.json())
         self._next_reference = values[0] if values and values[0] else None
 
+        if (
```

Comment:
`has_more` handling needs to be aligned with `RangePaginator`. Keep the handling here for now, we will refactor it in a follow-up PR. Here you have JSONPath lookup three times consecutively and missing error handling when `has_more_path` is provided but the field is missing.